.gitlab-ci.yml

```diff
@@ -40,10 +45,20 @@ build:raspbian_latest:
     - build_prep
   services:
     - docker:dind
+  variables:
+    CONTAINER_TAG: "rasbian_latest"
   script:
+    - echo "INFO - Building and Pushing ${CONTAINER_TAG}-${CI_PIPELINE_ID} to the registry ${CI_REGISTRY_IMAGE}"
+    - echo $CI_REGISTRY_PASSWORD | docker login -u $CI_REGISTRY_USER $CI_REGISTRY --password-stdin
     - tar -xvf docker-files-raspbian.tar
-    - docker build --build-arg raspbian_version=${RASPBIAN_VERSION} -t $DOCKER_REPOSITORY:raspbian_latest docker-files-raspbian
-    - docker save $DOCKER_REPOSITORY:raspbian_latest > raspbianImage.tar
+    - docker build 
+        --cache-from ${CI_REGISTRY_IMAGE}:${CONTAINER_TAG}
```

Comment:
I think this is wrong. The final images now are `${CI_REGISTRY_IMAGE}:${CONTAINER_TAG}-master` so we should rather use that as a cache source.

Nothing will be updating `${CI_REGISTRY_IMAGE}:${CONTAINER_TAG}` any longer after this PR.

Reply:
Right! Let me fix that